packages/web-components/src/wallet-connection/walletConnection.js

```diff
@@ -1,13 +1,78 @@
 // @ts-check
-import { getKeplrAddress } from '../getKeplrAddress.js';
+import { SigningStargateClient } from '@cosmjs/stargate';
+import { subscribeLatest } from '@agoric/notifier';
+import { makeInteractiveSigner } from './makeInteractiveSigner.js';
 import { watchWallet } from './watchWallet.js';
+import { Errors } from '../errors.js';
 
 export const makeAgoricWalletConnection = async chainStorageWatcher => {
-  const address = await getKeplrAddress(chainStorageWatcher.chainId);
+  if (!('keplr' in window)) {
+    throw Error(Errors.noKeplr);
+  }
+  /** @type {import('@keplr-wallet/types').Keplr} */
+  // @ts-expect-error cast (checked above)
+  const keplr = window.keplr;
+
+  const { address, submitSpendAction } = await makeInteractiveSigner(
+    chainStorageWatcher.chainId,
+    chainStorageWatcher.rpcAddr,
+    keplr,
+    SigningStargateClient.connectWithSigner,
+  );
 
   const walletNotifiers = await watchWallet(chainStorageWatcher, address);
 
+  const makeOffer = async (
+    invitationSpec,
+    proposal,
+    offerArgs,
+    onStatusChange,
+    id = new Date().getTime(),
+  ) => {
+    const { marshal } = chainStorageWatcher;
+    const spendAction = marshal.serialize(
+      harden({
+        method: 'executeOffer',
+        offer: {
+          id,
+          invitationSpec,
+          proposal,
+          offerArgs,
+        },
+      }),
+    );
+
+    try {
+      // eslint-disable-next-line @jessie.js/safe-await-separator
```

Comment:
try an `await null` before the this `try{}`. the lint rule shouldn't complain

Reply:
Done, I just wasn't sure that was any better.